### PR TITLE
filtering same regions across all samples

### PR DESCRIPTION
**Pull request by Young-Sook**
CRADLE used to filter out the regions with the total coverage less than `mi` across all samples and then filters regions that does not have coverage more than `mi/the number of samples` in a sample-specific way. This approach could lead to a false positive by performing statistical testing to the regions where more than one sample have no corrected read counts (`np.nan`) and treating those values equal to zero. 

In this version, CRADLE filters out the same regions across all samples. It filters out 
1) regions with the total coverage less than `mi` 
2) regions where more than one sample have less coverage than  `mi/the number of samples`

### CODE REVIEW

**Review by tncowart — COMMENTED** [review]
I have a few suggestions and suggested changes. Also, instead of pressing "Commit suggestion" please either press "Add suggestion to batch" (so you don't make quite so many commits) or make all the changes yourself locally and commit them all at once (not bothering with "Commit suggestion"/"Add suggestion to batch" stuff).

Thank you!

File: CRADLE/CorrectBiasStored/correctReadCounts.py [hunk 1]
```diff
@@ -137,18 +136,18 @@ def selectOverallIdx(chromo, analysisStart, analysisEnd, bwFiles, minFragFilterV
 			readCounts = np.array(bwFile.values(chromo, analysisStart, analysisEnd), dtype=np.float32)
 		readCounts[np.isnan(readCounts)] = 0.0
 
+		currIdx = np.where(readCounts > meanMinFragFilterValue)[0]
+		currIdx = np.intersect1d(prevIdx, currIdx)
+		prevIdx = currIdx
```
[the remark this answers is not in the document]
Comment (Young-Sook) [reply CRADLE/CorrectBiasStored/correctReadCounts.py:141]: what do we need replicateIdx?

File: CRADLE/CorrectBias/readCounts.pyx [hunk 2]
```diff
@@ -126,8 +124,6 @@ cpdef correctReadCounts(covariFileName, chromo, analysisStart, analysisEnd, last
 def selectIdx(chromo, regionStart, regionEnd, ctrlBWNames, experiBWNames, lastBinStart, lastBinEnd, nBins, binSize, highRC, minFragFilterValue):
 	meanMinFragFilterValue = int(np.round(minFragFilterValue / (len(ctrlBWNames) + len(experiBWNames))))
 
-	ctrlSpecificIdx = []
-	experiSpecificIdx = []
 	for rep, bwName in enumerate(ctrlBWNames):
```
[the remark this answers is not in the document]
Comment (Young-Sook) [reply CRADLE/CorrectBias/readCounts.pyx:127]: Sure, then we need to merge several things from control and expbw —for example, merging`vari.COEFEXP_HIGHRC` with `vari.COEFCTRL_HIGHRC`. I think that is the scope of another PR that changes the whole structure. In this PR, I focused on changing the filtering strategy.

File: CRADLE/CorrectBias/readCounts.pyx [hunk 3]
```diff
@@ -173,23 +171,19 @@ def selectIdx(chromo, regionStart, regionEnd, ctrlBWNames, experiBWNames, lastBi
...
 	idx = np.where(rc_sum > minFragFilterValue)[0].tolist()
-	if len(idx) == 0:
-		ctrlSpecificIdx = [ [] for _ in range(len(ctrlBWNames)) ]
-		experiSpecificIdx = [ [] for _ in range(len(experiBWNames)) ]
+	overallIdx = np.intersect1d(idx, currIdx)
+
+	if len(overallIdx) == 0:
 		highReadCountIdx = []
-		return ctrlSpecificIdx, experiSpecificIdx, highReadCountIdx
+		return overallIdx, highReadCountIdx
```
[the remark this answers is not in the document]
Comment (Young-Sook) [reply CRADLE/CorrectBias/readCounts.pyx:185]: yes, but I just wanted to make what `[]` means there clear.

File: CRADLE/CorrectBiasStored/correctReadCounts.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tncowart) [reply CRADLE/CorrectBiasStored/correctReadCounts.py:141]: it's just a name change as part of the three "suggested changes" here. I think it better represents the meaning of the index than "prevIdx" or "currIdx".

File: CRADLE/CorrectBias/readCounts.pyx (hunk 2 above)
[the remark this answers is not in the document]
Comment (tncowart) [reply CRADLE/CorrectBias/readCounts.pyx:127]: You don't need to do all that just to change this function's internals. It doesn't use any of that. Can't you change line 127 to `for rep, bwName in enumerate(ctrlBWNames + experiBWNames):` and delete the `for` loop that starts on line 156?

File: CRADLE/CorrectBiasStored/correctReadCounts.py [hunk 4]
```diff
@@ -137,18 +136,18 @@ def selectOverallIdx(chromo, analysisStart, analysisEnd, bwFiles, minFragFilterV
...
 	idx = np.where(readCountSums > minFragFilterValue)[0]
+	idx = np.intersect1d(idx, currIdx)
 
 	return idx
 
 def selectHighRCIdx(trainingReadCounts, idx, highRC):
 	highReadCountIdx = np.where(trainingReadCounts > highRC)[0]
 	highReadCountIdx = np.intersect1d(highReadCountIdx, idx)
 	return highReadCountIdx
-
```
[the remark this answers is not in the document]
Comment (Young-Sook) [reply CRADLE/CorrectBiasStored/correctReadCounts.py:150]: I just added it

File: CRADLE/CorrectBias/readCounts.pyx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Young-Sook) [reply CRADLE/CorrectBias/readCounts.pyx]: Agree. I changed as you suggested.

**Pull request merged**